pennylane/estimator/templates/select.py

```diff
@@ -166,12 +213,23 @@ def resource_rep(
             raise TypeError(
                 f"`rotation_precision` must be an integer, but type {type(rotation_precision)} was provided."
             )
+
+        if batched_rotations is not None and (
+            not isinstance(batched_rotations, int)
+            or batched_rotations <= 0
+            or batched_rotations > thc_ham.num_orbitals - 1
+        ):
```

Comment:
```suggestion
        if batched_rotations is not None and (
            not isinstance(batched_rotations, int)
            or batched_rotations not in range(1, thc_ham.num_orbitals)
        ):
```

might not be formatted correctly. Also change this in general.